src/task4.py

```diff
@@ -6,7 +6,9 @@
 import traceback
 from datetime import datetime
 
-class decorator_4:
+from task3 import decorator_3
+
+class decorator_4(decorator_3):
```

Comment:
Your `decorator_4` previously was only calling the function and reporting error. According to the problem statement, it should also extend all the functionalities of the previous decorators, that's why I added the inheritance from `decorator_3` to access those functionalities as well.